wallet-standard/cryptid-wallet/src/wallet.ts

```diff
@@ -0,0 +1,272 @@
+            },
+        };
+    }
+
+    get accounts() {
+        return this.#account ? [this.#account] : [];
+    }
+
+    constructor(cryptidWallet: CryptidWallet) {
+        if (new.target === CryptidWalletWallet) {
+            Object.freeze(this);
+        }
+
+        this.#cryptidWallet = cryptidWallet;
+
+        cryptidWallet.on('connect', this.#connected, this);
+        cryptidWallet.on('disconnect', this.#disconnected, this);
+        cryptidWallet.on('accountChanged', this.#reconnected, this);
+
+        this.#connected();
+    }
+
+    #on: EventsOnMethod = (event, listener) => {
+        this.#listeners[event]?.push(listener) || (this.#listeners[event] = [listener]);
+        return (): void => this.#off(event, listener);
+    };
+
+    #emit<E extends EventsNames>(event: E, ...args: Parameters<EventsListeners[E]>): void {
+        // eslint-disable-next-line prefer-spread
+        this.#listeners[event]?.forEach((listener) => listener.apply(null, args));
+    }
+
+    #off<E extends EventsNames>(event: E, listener: EventsListeners[E]): void {
+        this.#listeners[event] = this.#listeners[event]?.filter((existingListener) => listener !== existingListener);
+    }
+
+    #connected = () => {
+        const address = this.#cryptidWallet.publicKey?.toBase58();
+        if (address) {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const publicKey = this.#cryptidWallet.publicKey!.toBytes();
+
+            const account = this.#account;
+            if (!account || account.address !== address || !bytesEqual(account.publicKey, publicKey)) {
+                this.#account = new CryptidWalletWalletAccount({ address, publicKey });
+                this.#emit('change', { accounts: this.accounts });
+            }
+        }
+    };
+
+    #disconnected = () => {
+        if (this.#account) {
+            this.#account = null;
+            this.#emit('change', { accounts: this.accounts });
+        }
+    };
+
+    #reconnected = () => {
+        if (this.#cryptidWallet.publicKey) {
+            this.#connected();
+        } else {
+            this.#disconnected();
+        }
+    };
+
+    #connect: ConnectMethod = async ({ silent } = {}) => {
+        if (!this.#account) {
+            await this.#cryptidWallet.connect(silent ? { onlyIfTrusted: true } : undefined);
+        }
+
+        this.#connected();
+
+        return { accounts: this.accounts };
+    };
+
+    #disconnect: DisconnectMethod = async () => {
+        await this.#cryptidWallet.disconnect();
+    };
+
+    #signAndSendTransaction: SolanaSignAndSendTransactionMethod = async (...inputs) => {
+        if (!this.#account) throw new Error('not connected');
+
+        const outputs: SolanaSignAndSendTransactionOutput[] = [];
+
+        if (inputs.length === 1) {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const { transaction, account, chain, options } = inputs[0]!;
```

Comment:
I don't think the non null assertion is necessary? We should never use non null in non tests as a general rule.

It looks like on `accounts` is used here?